### DISCUSSION TITLE
NVIDIA Shield Support

### DISCUSSION THREAD

**Discussion by PRProd** [PRProd/HA-Firemote#52] [Announcements]
# NVIDIA Shield Support is here!

## New features
 - Native support for NVIDIA Shield devices!
 - Remote Style Override: Regardless of the kind of device you own, you can switch your Firemote card to look like a completely different remote control.  If you were very clever, you probably knew of a way to do this already, but now it's built in!
 - Visible Device Name Color Choice: As requested in #44, you can now choose a different color for the visible device name text. 
 - Sometimes apps made for the Amazon Fire world weren't the same as the apps made for other Android TV style devices which needed to be sorted out.  If you've used Firemote for any non Amazon Fire devices, you may have come across this problem before, as @moh-Brent discovered in #30. These issues have been all sorted out under the hood which will absolutely result in a better experience

## New App Shortcuts
 - OQEE by Free #39
 - TiVimate #32, #48
 - SmartTubeNext (STN) #48
 - Sky News #49
 - IPVanish #49
 - VTM GO (BE) #50
 - Streamz (BE) #50
 - VRT MAX (BE) #50
 - Freevee

## Thank you!
I'd like to give a HUGE thank you to @MRobi1 & @jrspowers for their help as beta testers! #47

<hr /><em>This discussion was created from the release <a href='https://github.com/PRProd/HA-Firemote/releases/tag/v2.0.0'>NVIDIA Shield Support</a>.</em>

**Comment by jrspowers** [comment]
hello @PRProd the addition for 6 apps shortcuts on the nvidia remote will it come later?  because you have now released the full version?

**Reply by PRProd** [reply]
It's in the production release

**Reply by jrspowers** [reply]
> It's in the production release

can i test this already?

**Reply by PRProd** [reply]
It's available for everybody to use

**Reply by jrspowers** [reply]
> It's available for everybody to use

i can only set 4 apps shortcuts on nvidia app i would like to set 6 but that is not possible i updated to 2.0.1 but is not possible?

**Reply by jrspowers** [reply]
> It's available for everybody to use

work after re-downloading the latest version

**Reply by PRProd** [reply]
@jrspowers - If you're still super excited about having a ton of app launch buttons, I just put out a new pre-release which introduces functionality for the Xiaomi Mi Box.  One style of the Mi Box remote is available to use on any device with up to 10 app launch buttons!

**Reply by jrspowers** [reply]
> @jrspowers - If you're still super excited about having a ton of app launch buttons, I just put out a new pre-release which introduces functionality for the Xiaomi Mi Box. One style of the Mi Box remote is available to use on any device with up to 10 app launch buttons!

hello @PRProd  this all available on the xiomi mi remote & not on the shield remote so i see would it be helpful if this comes on shield remote ?  can also do the xioami mi box test for you later, I still have 1 at home, it will be back in startup later

**Reply by jrspowers** [reply]
@PRProd can you add this icon to vtm go would also make it beautiful ?  thanks in advance! 

[vtm go logo](https://images.app.goo.gl/mAHgkkJX8eqJGbV16)

**Reply by jrspowers** [reply]
> > @jrspowers - If you're still super excited about having a ton of app launch buttons, I just put out a new pre-release which introduces functionality for the Xiaomi Mi Box. One style of the Mi Box remote is available to use on any device with up to 10 app launch buttons!
> 
> hello @PRProd  this all available on the xiomi mi remote & not on the shield remote so i see would it be helpful if this comes on shield remote ?  can also do the xioami mi box test for you later, I still have 1 at home, it will be back in startup later


will this also come for nvidia shield remote?

**Reply by PRProd** [reply]
No.  I will not add that many to that remote.  Six is still too many in my opinion, ten would make it too ridiculously tall.  If you do want all 10, did you know that you can configure your Firemote to use the look of the Xiaomi Mi remote even though it's controlling a Shield?  Just change the value of "Remote Style" like this:  (currently only available in the latest Pre Release version #55 )
![image](https://user-images.githubusercontent.com/25486599/228044705-0bcf6fe5-d7e3-48b3-9788-e44e02b0fb6e.png)

**Reply by PRProd** [reply]
> @PRProd can you add this icon to vtm go would also make it beautiful ? thanks in advance!
> 
> [vtm go logo](https://images.app.goo.gl/mAHgkkJX8eqJGbV16)

I need a .svg file with a very small file size.  I wasn't able to quickly find one yesterday, so I moved on.  I'll look some more later.  Unfortunately, the one you linked to is a .png file.

**Reply by jrspowers** [reply]
> No. I will not add that many to that remote. Six is still too many in my opinion, ten would make it too ridiculously tall. If you do want all 10, did you know that you can configure your Firemote to use the look of the Xiaomi Mi remote even though it's controlling a Shield? Just change the value of "Remote Style" like this:

okay thanks for the info definitely go check it out & the xiaomi mi remote works perfectly on xiaomi mi box everything works great & fast!  I think this one can be released!

**Reply by jrspowers** [reply]
> > @PRProd can you add this icon to vtm go would also make it beautiful ? thanks in advance!
> > [vtm go logo](https://images.app.goo.gl/mAHgkkJX8eqJGbV16)
> 
> I need a .svg file with a very small file size. I wasn't able to quickly find one yesterday, so I moved on. I'll look some more later. Unfortunately, the one you linked to is a .png file.

hello, i just looked at the logo and converted it to svg, it is 1kb in size so really very small hopefully you can use it? i have 2 version 1 with black background & with white background

[vtm go logo(black background](https://mega.nz/file/WxwmEbRQ#iKC6KoSziiX5m_myTm9rqermLfqo74Hb5dRfYTNwWMc)
[vtm go logo white background](https://mega.nz/file/bhZQhTgb#3M3S5ZYCCDvYTUZHUIux7IiTageX9wCOl12kVyIR2Bs)

**Reply by ChristophCaina** [reply]
may I ask, where I can find the "MI" Remote style?
I only have the following available in the selection:

![image](https://user-images.githubusercontent.com/26391061/228051879-00c852e8-1a64-453f-9e52-aa101d8076c9.png)

**Reply by PRProd** [reply]
@ChristophCaina - You first need to have the latest pre-release version of Firemote installed.  As of right now, that functionality is in beta testing.  Feel free to join the beta testing discussion!  #55

**Reply by jrspowers** [reply]
> No.  I will not add that many to that remote.  Six is still too many in my opinion, ten would make it too ridiculously tall.  If you do want all 10, did you know that you can configure your Firemote to use the look of the Xiaomi Mi remote even though it's controlling a Shield?  Just change the value of "Remote Style" like this:  (currently only available in the latest Pre Release version #55 )
> 
> ![image](https://user-images.githubusercontent.com/25486599/228044705-0bcf6fe5-d7e3-48b3-9788-e44e02b0fb6e.png)
> 
> 

what's the name of this button wall this can't be hidden try something but don't get any of the listed work I thought about app-switch-button but that's not it so can you tell me this?

![image](https://user-images.githubusercontent.com/60328474/228070340-4b361d51-c31f-4979-86ac-fb46ce8170fc.jpeg)

**Reply by PRProd** [reply]
@jrspowers - That button is called "apps-button".  Please remind me to add that to the official launch documentation if I forget.
```yaml
button_overrides:  
  apps-button:
    hidden: true
```

**Reply by jrspowers** [reply]
Thanks man 😊

**Comment by PRProd** [comment]
@MRobi1 - You might be interested in the new [v2.0.4](https://github.com/PRProd/HA-Firemote/releases/tag/v2.0.4) that I just pushed out.  I added some functions into the app list, including the ability to open the "app switcher" function like we were talking about.

**Reply by jrspowers** [reply]
the very nice addition that app switcher could this also be set to e.g. the keyboard button?

**Reply by MRobi1** [reply]
Works great!! I wonder if the icon should be worded as "Switch Apps"? App Switch, while it's the name of the function, sounds backwards when said out loud lol

@jrspowers I believe he mentioned a limitation in capturing double press or hold on the buttons with the way it is coded, which may limit it to being assigned to other buttons. The way it works on the regular shield remote is to double press the home button. Having it as an app button is a good workaround I think.

**Reply by PRProd** [reply]
@MRobi1 - Sure thing!  That change will come with the next release.  My brain made me write it that way because the underlying command is called "KEYCODE_APP_SWITCH", but you're right, it sounds better as "Switch Apps"

@jrspowers - @MRobi1 is correct, I haven't yet found a way (if it's even possible) to properly capture a long press or double press using the language framework built into Home Assistant cards.  However, using the button override functionality, you can accomplish this through a separate script, or through a simple YAML config.  Here is an example of a YAML config that would accomplish what you are asking for:
```yaml
button_overrides:
  keyboard-button:
    service: androidtv.adb_command
    data:
      command: adb shell input keyevent KEYCODE_APP_SWITCH
    target:
      entity_id: media_player.android_tv_192_168_107_209
```
